src/vs/workbench/contrib/search/browser/searchModel.ts

```diff
@@ -1946,6 +1975,103 @@ export class SearchModel extends Disposable {
 		return this._searchResult;
 	}
 
+	private async runFileQueries(includes: (string)[], token: CancellationToken, textQuery: ITextQuery): Promise<URI[]> {
+		const promises = includes.map(include => {
+			const query: IFileQuery = {
+				type: QueryType.File,
+				filePattern: include,
+				folderQueries: textQuery.folderQueries
+			};
+			return this.searchService.fileSearch(
```

Comment:
How many file searches does this trigger? One per notebook type?

Reply:
Does this apply the max result count by default?

Reply:
iirc, yes, it is once per notebook. Each of them provide their own glob pattern. I was originally planning on concatenating all the globs into one glob using brace expansion, but ripgrep doesn't support nested braces in glob patterns.

Reply:
Maybe there's a smart way to parse the globs and build one single pattern than matches the others

Reply:
The most straightforward way of doing this would likely be with using braces (`{ globA, globB, globC }`), but if the individual globs have braces in them, ripgrep won't support this because it doesn't support nested alternate groups. 
<img width="678" alt="image" src="https://github.com/microsoft/vscode/assets/31675041/f9d2145d-e630-4492-82e2-37a1703ee41a">

We could perform brace expansion, but I was thinking that it'd be an optimization for later.